src/components/molecules/AddToDashboard.tsx

```diff
@@ -15,7 +15,7 @@ type Props = {
 const AddToDashboard: React.FC<Props> = (props) => {
     const history = useHistory();
 
-    const key = `${props.dashboardItem.dataSourceId}-${props.dashboardItem.visualisationType}` as VisualisationMappingFunctionPath;
+    const key = props.dashboardItem.id ?? uuidv4();
```

Comment:
I think it might be beneficial if this doesn't change when dashboardItem changes, and only when it's (re)mounted.

You can solve this with a `useMemo`.